tensorboard/webapp/plugins/npmi/store/npmi_reducers_test.ts

```diff
@@ -268,4 +268,485 @@ describe('npmi_reducers', () => {
       });
     });
   });
+
+  describe('Annotation Selection', () => {
+    it('select annotations without duplicates', () => {
+      const state = createNpmiState({
+        selectedAnnotations: ['annotation_1'],
+      });
+      const nextState = reducers(
+        state,
+        actions.addSelectedAnnotations({
+          annotations: ['annotation_1', 'annotation_2'],
+        })
+      );
+      expect(nextState.selectedAnnotations).toEqual([
+        'annotation_1',
+        'annotation_2',
+      ]);
+    });
+
+    it('remove a selected annotation', () => {
+      const state = createNpmiState({
+        selectedAnnotations: ['annotation_1', 'annotation_2'],
+      });
+      const nextState = reducers(
+        state,
+        actions.removeSelectedAnnotation({annotation: 'annotation_1'})
+      );
+      expect(nextState.selectedAnnotations).toEqual(['annotation_2']);
+    });
+
+    it('set the selected annotations', () => {
+      const state = createNpmiState({
+        selectedAnnotations: ['annotation_1', 'annotation_2'],
+      });
+      const nextState = reducers(
+        state,
+        actions.setSelectedAnnotations({
+          annotations: ['annotation_1', 'annotation_3'],
+        })
+      );
+      expect(nextState.selectedAnnotations).toEqual([
+        'annotation_1',
+        'annotation_3',
+      ]);
+    });
+
+    it('clear selected annotations', () => {
+      const state = createNpmiState({
+        selectedAnnotations: ['annotation_1', 'annotation_2'],
+      });
+      const nextState = reducers(state, actions.clearSelectedAnnotations());
+      expect(nextState.selectedAnnotations).toEqual([]);
+    });
+  });
+
+  describe('Annotation Flagging', () => {
+    it('flag annotations with no annotation flagged', () => {
+      const state = createNpmiState();
+      const nextState = reducers(
+        state,
+        actions.flagAnnotations({
+          annotations: ['annotation_1', 'annotation_2', 'annotation_3'],
+        })
+      );
+      expect(nextState.flaggedAnnotations).toEqual([
+        'annotation_1',
+        'annotation_2',
+        'annotation_3',
+      ]);
+    });
+
+    it('flagging annotations with some already flagged', () => {
```

Comment:
say something about not changing the order of the existing flag annotation if that is part of the contract (or add a new spec that, even if it looks similar/same, codifies the contract)